proptest/src/num/float_samplers.rs

```diff
@@ -0,0 +1,338 @@
+                    let high = high.borrow().0;
+
+                    let values = SampleValueCollection::new_inclusive(low, last_float(high));
+
+                    FloatUniform {
+                        uniform: Uniform::new(0, values.count),
+                        values,
+                    }
+                }
+
+                fn new_inclusive<B1, B2>(low: B1, high: B2) -> Self
+                where
+                    B1: SampleBorrow<Self::X> + Sized,
+                    B2: SampleBorrow<Self::X> + Sized,
+                {
+                    let low = low.borrow().0;
+                    let high = high.borrow().0;
+
+                    let values = SampleValueCollection::new_inclusive(low, high);
+
+                    FloatUniform {
+                        uniform: Uniform::new(0, values.count),
+                        values,
+                    }
+                }
+
+                fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Self::X {
+                    $wrapper(self.values.get(self.uniform.sample(rng)))
+                }
+            }
+
+            impl SampleUniform for $wrapper {
+                type Sampler = FloatUniform;
+            }
+
+            #[derive(Clone, Copy, Debug)]
+            struct SampleValueCollection {
+                start: $typ,
+                end: $typ,
+                step: $typ,
+                count: $int_typ,
+            }
+
+            // Values greater than MAX_PRECISE_INT may be rounded when converted to float.
+            const MAX_PRECISE_INT: $int_typ =
+                (2 as $int_typ).pow($typ::MANTISSA_DIGITS);
+
+            // The collection of sample values that may be generated by UniformF32U.
+            impl SampleValueCollection {
+                fn new_inclusive(low: $typ, high: $typ) -> Self {
+                    assert!(low.is_finite(), "low finite");
+                    assert!(high.is_finite(), "high finite");
+                    assert!(high - low >= 0., "invalid range");
+
+                    let min_abs = $typ::min(low.abs(), high.abs());
+                    let max_abs = $typ::max(low.abs(), high.abs());
+
+                    let gap = last_gap(max_abs);
+
+                    let (start, end, step) = if low.abs() < high.abs() {
+                        (high, low, -gap)
+                    } else {
+                        (low, high, gap)
+                    };
+
+                    let min_gaps = min_abs / gap;
+                    let max_gaps = max_abs / gap;
+                    debug_assert!(
+                        max_gaps.floor() == max_gaps,
+                        "max_gaps is an integer"
+                    );
+
+                    let count = if low.signum() == high.signum() {
+                        max_gaps as $int_typ - min_gaps.floor() as $int_typ
+                    } else {
+                        max_gaps as $int_typ + min_gaps.ceil() as $int_typ
+                    } + 1;
```

Comment:
i feel like there's an obvious reason for this, but i can't seem to piece together why we add `1` at the end here, mind explaining?

Reply:
right, thank you